src/babashka/process.cljc

```diff
@@ -178,17 +178,28 @@
       (str/includes? "windows")))
 
 (defn- -program-resolver [{:keys [program dir]}]
-  ;; this should make life easier and not cause any bugs that weren't there previously
-  ;; on exception we just return the program as is
-  (try
-    (if (fs/relative? program)
-      (if-let [f (fs/which (if dir
-                             (-> (fs/file dir program) fs/absolutize)
-                             program))]
-        (str f)
-        program)
-      program)
-    (catch Throwable _ program)))
+  ;; Used by default:
+  ;; - for all program resolution when running on Windows
+  ;; - by `exec` on all OSes
+  ;; Default ProcessBuilder behaviour on macOS/Linux does this work naturally
+  ;; and is therefore unneeded.
+  ;;
+  ;; ProcessBuilder program resolution on Windows does not entirely match
+  ;; behaviour of CMD Shell nor PowerShell.
+  ;; Here we adapt resolution to match PowerShell (with the exclusion of resolving
+  ;; .ps1 scripts) which follows the same strategy as macOS/Linux.
+  (if-let [resolved (cond
+                      (fs/absolute? program)
+                      (fs/which program) ;; to resolve any extensions
+
+                      (fs/parent program)
+                      (fs/which (fs/file (or dir (fs/cwd))
+                                         program))
+
+                      :else
+                      (fs/which program))]
+    (-> resolved fs/absolutize fs/normalize str)
```

Comment:
Why is this change necessary? It wasn't there before?

Reply:
We did absolutize before but only for relative programs with a dir.
(This was a change I made a while back f2c8d93c921eefe2344cf8adec004b5f28b272f6, I am now wondering if it _absolutely_ necessary? :drum:)

It could be I am over-absolutizing here. The thing I liked about it is that it makes it very clear exactly what we are resolving to. I think I added normalize to make it clearer what we were absolutizing to, and that also made some test comparisons easier.

But I also see the value of not absolutizing/normalizing unless we need to. Can explore.

Reply:
I prefer not processing the program name unless we have to as this may also affect the name of the program that is seen in exceptions etc. (same reason I don't prefer adding an extra `cwd` above)

Reply:
Sounds reasonable to me, I'll make the change.

Reply:
I committed the first suggested (because all the tests still worked, I felt it was permissible) but this other change fails a bunch of tests. To diagnose via CI I reversed the bb and JVM tests as the JVM tests give better error messages. Then when I changed the `resolved` return value to `(str resolved)` half the CI tests started working, except Windows. FWIW.